src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -186,6 +253,11 @@ public Map<String, Object> getProcessedRecordForStreamingIngest(SinkRecord recor
         e.printStackTrace();
       }
     }
+
+    if (hasCaseInsensitiveDuplicates(streamingIngestRow)) {
+      throw SnowflakeErrors.ERROR_0025.getException();
+    }
+
```

Comment:
discussed offline, we don't need to do this anymore, we could have a function called `formatColumnName` which convert everything to uppercase if no double quotes, please take a look at how [ingest SDK](https://github.com/snowflakedb/snowflake-ingest-java/blob/83008cd9259fb18518f3167261aacdc664bdb992/src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java#L617) is handling it

Reply:
And could you add a test on double quotes column name?